lib/linkerd2-proxy-core/src/listen.rs

```diff
@@ -1,27 +1,39 @@
-use crate::drain;
-use futures::Future;
+use futures::{Future, Poll};
 use linkerd2_error::Error;
+use tower::Service;
 
-pub trait ListenAndSpawn {
+pub trait Listen {
```

Comment:
This trait seems semantically similar to `Stream<Item = Connection>` — should there be a blanket impl or something?

Reply:
I'm not opposed to it, but we have no use for it at the moment. Less code wins ;)